### PR TITLE
Remove order creation single-selection and feature flag

### LINKED ISSUE #9081: [Product Multi-Selection] Remove feature flag
Issue opened so we don't forget to clear up unused flags. 

Remove `productMultiSelectionM1` when needed.

### LINKED ISSUE #9452: [Product Multi-Selection] Remove single-selection logic
The logic for single-selection is still in place, as we wanted to be sure to tackle both cases. Opening this issue to track its removal as part of paying the project's technical debt.

### PR DESCRIPTION
**Pull request by iamgabrielma**
Closes: #9081 
Closes: #9452 

## Description
This PR addresses removing the legacy logic for product single-selection by:
- Removes feature flag from `DefaultFeatureFlagService`
- Removes the flag from `MockFeatureFlagService`
- Removes order creation single-selection logic
- Updates Unit Tests

## Testing instructions
There are no changes in the logic compared with the feature flag being enabled, but the following can be confirmed:
- Go to Orders > Tap `+` > Confirm that the string `+ Add Products` it's plural, then tap on it
- Products can be selected/unselected on a multi-selection basis, and tapping on "X products selected" does add them to the Order.
- On an Order with products, these can be removed  by tapping on `-` , and confirming by tapping on `Delete product from order`.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift [hunk 1]
```diff
@@ -180,7 +174,7 @@ final class EditableOrderViewModel: ObservableObject {
             purchasableItemsOnly: true,
             storageManager: storageManager,
             stores: stores,
-            supportsMultipleSelection: isProductMultiSelectionEnabled,
+            supportsMultipleSelection: true,
```
Comment (toupper) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift:177]: Now that we removed single-selection, can we remove this parameter/property from the initializer/class?

**Review by toupper — APPROVED** [review]
LGTM and tests well, just a comment about some more code we can remove.

**Comment by toupper** [comment]
Nice job @iamgabrielma ! One question though, we still have two almost identical initializers, do we have plans to remove one of them?

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift (hunk 1 above)
Comment (iamgabrielma) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift:177]: Yes! As this bit would involve modifying the initializer we also use for Coupons, we're tracking this on a different issue: https://github.com/woocommerce/woocommerce-ios/issues/9434

**Comment by iamgabrielma** [comment]
Thanks for the review! 

> Nice job @iamgabrielma ! One question though, we still have two almost identical initializers, do we have plans to remove one of them?

Yes, I have a different issue opened here: https://github.com/woocommerce/woocommerce-ios/issues/9434 that will take care of the initializer duplication as well as removing the `supportsMultipleSelection` property. 

I decided to keep it separate because removing the property and the two initializers also involves modifying the selection state logic that is performed on the different callbacks when products are selected/unselected, as well as the initializer for product variations. That one will come up next! 👍

**Pull request merged**